compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1936,14 +1939,20 @@ class Typer extends Namer
       }
       var checkedArgs = preCheckKinds(args1, paramBounds)
         // check that arguments conform to bounds is done in phase PostTyper
-      if (tpt1.symbol == defn.andType)
+      val tycon = tpt1.symbol
+      if (tycon == defn.andType)
         checkedArgs = checkedArgs.mapconserve(arg =>
           checkSimpleKinded(checkNoWildcard(arg)))
-      else if (tpt1.symbol == defn.orType)
+      else if (tycon == defn.orType)
         checkedArgs = checkedArgs.mapconserve(arg =>
           checkSimpleKinded(checkNoWildcard(arg)))
+      else if tycon == defn.throwsAlias
+          && checkedArgs.length == 2
+          && checkedArgs(1).tpe.derivesFrom(defn.RuntimeExceptionClass)
```

Comment:
Is there really a gain in explicitly preventing users from declaring throwing `RuntimeException`s? And even if we disallow methods like
```scala
def foo(): Int throws ArithmeticException = ???
```
then
```scala
def foo()(using CanThrow[ArithmeticException]): Int = ???
```
will be still valid, right? So this somehow defeats the purpose IMHO

Reply:
We won't generate a capability for it, so it's better to communicate that early. But I think ding this for throws clauses is enough. I don't want to start imposing restrictions on general implicit parameters.